code/game/machinery/camera/camera_assembly.dm

```diff
@@ -236,6 +236,14 @@
 	forceMove(C)
 	C.setDir(src.dir)
 
+	// Add upgrades
+	if(xray_module)
+		C.upgrades += CAMERA_UPGRADE_XRAY
+	if(emp_module)
+		C.upgrades += CAMERA_UPGRADE_EMP_PROOF
+	if(proxy_module)
+		C.upgrades += CAMERA_UPGRADE_MOTION
+
```

Comment:
this should all be handled during the camera's init using the proper procs. is the assembly not being passed correctly? 

https://github.com/tgstation/tgstation/blob/c5e3d7a4e70784870250b06cc2df56f69197fbd3/code/game/machinery/camera/camera.dm#L64-L83